chemprop/utils.py

```diff
@@ -190,80 +202,171 @@ def load_frzn_model(model: torch.nn,
-            model_encoder_param_names = [item for sublist in model_encoder_param_names for item in sublist]
-            for loaded_param_name,model_param_name in zip(loaded_encoder_param_names,model_encoder_param_names):
-                model_state_dict = overwrite_state_dict(loaded_param_name,model_param_name,loaded_state_dict,model_state_dict)
-        
-        if current_args.frzn_ffn_layers > 0: # Duplicate encoder from frozen checkpoint and overwrite all encoders + FFN layers
-            raise Exception ('Number of molecules in checkpoint_frzn must be equal to current model for ffn layers to be frozen')
-            
-    elif (loaded_args.number_of_molecules>1 )& (current_args.number_of_molecules>1):
-        if (loaded_args.number_of_molecules) !=( current_args.number_of_molecules):
-            raise Exception('Number of molecules in checkpoint_frzn ({}) must match current model ({}) OR equal to 1.'.format(loaded_args.number_of_molecules,current_args.number_of_molecules))
-        
+                model_state_dict = overwrite_state_dict(
+                    param_name, param_name, loaded_state_dict, model_state_dict
+                )
+        if (
+            current_args.checkpoint_frzn is not None
+            and not current_args.freeze_first_only
+            and current_args.frzn_ffn_layers <= 0
+        ):  # Duplicate encoder from frozen checkpoint and overwrite all encoders
+            loaded_encoder_param_names = [
+                "encoder.encoder.0.W_i.weight",
+                "encoder.encoder.0.W_h.weight",
+                "encoder.encoder.0.W_o.weight",
+                "encoder.encoder.0.W_o.bias",
+            ] * current_args.number_of_molecules
+            model_encoder_param_names = [
+                [
+                    (
+                        f"encoder.encoder.{mol_num}.W_i.weight",
+                        f"encoder.encoder.{mol_num}.W_h.weight",
+                        f"encoder.encoder.{mol_num}.W_o.weight",
+                        f"encoder.encoder.{mol_num}.W_o.bias",
+                    )
+                ]
+                for mol_num in range(current_args.number_of_molecules)
+            ]
+            model_encoder_param_names = [
+                item for sublist in model_encoder_param_names for item in sublist
+            ]
+            for loaded_param_name, model_param_name in zip(
+                loaded_encoder_param_names, model_encoder_param_names
+            ):
+                model_state_dict = overwrite_state_dict(
+                    loaded_param_name, model_param_name, loaded_state_dict, model_state_dict
+                )
+
+        if (
+            current_args.frzn_ffn_layers > 0
+        ):  # Duplicate encoder from frozen checkpoint and overwrite all encoders + FFN layers
```

Comment:
Making this multiline just because of the trailing comment seems unnecessary
```suggestion
        # Duplicate encoder from frozen checkpoint and overwrite all encoders + FFN layers
        if current_args.frzn_ffn_layers > 0: 
```

Reply:
I'm of the opinion that comments are a stop-gap measure to make illegible code slightly more legible. You could just formally code the logic into a variable and make the code as a whole more readable while removing the comment itself. after all, characters are free:
```
duplicate_encoder = current_args.frzn_ffn_layers > 0
if duplicate_encoder:
    ...
```
there's the broader question of: "does it matter for a reader to understand the original, logical intention of this block?" This comment looks more like an internal note from the original author than anything else.

Reply:
I think that "duplicate" is a verb rather than an adjective. It's not saying that if there's a duplicate encoder then do this. It's saying that this is the block where we duplicate the encoder under these circumstances. That makes it feel like a more normal header comment before a block of code rather than an explanation of what the logic check is. I'm good with removing it entirely though as it doesn't feel critical here.

Reply:
two options:
1. clear up language so we're not using a word that both a verb or an adjective. should probably use something like "copy"
2. remove the comment and not deal with this

Reply:
Option 2 is nice and clean I'm on board with it